File: libgomp/testsuite/libgomp.oacc-fortran/lib-6.f90
! { dg-do run }

program main
  implicit none
  include "openacc_lib.h"

  integer n

  if (acc_get_num_devices (acc_device_host) .ne. 1) call abort

  if (acc_get_num_devices (acc_device_none) .ne. 0) call abort

  call acc_init (acc_device_host)

  if (acc_get_device_type () .ne. acc_device_host) call abort

  call acc_set_device_type (acc_device_host)

  if (acc_get_device_type () .ne. acc_device_host) call abort

  n = 0

  call acc_set_device_num (n, acc_device_host)

  if (acc_get_device_num (acc_device_host) .ne. 0) call abort

  if (.NOT. acc_async_test (n) ) call abort

  call acc_wait (n)

  call acc_wait_all ()

  call acc_shutdown (acc_device_host)

end program

File: libgomp/testsuite/libgomp.oacc-c-c++-common/lib-6.c
/* { dg-do run } */

#include <stdlib.h>
#include <openacc.h>

int
main (int argc, char **argv)
{
  int devnum;

  if (acc_get_device_type () == acc_device_default)
    abort ();

  if (acc_get_num_devices (acc_device_nvidia))
    {
      acc_set_device_type (acc_device_nvidia);

      if (acc_get_device_type () != acc_device_nvidia)
	abort ();

      acc_shutdown (acc_device_nvidia);

      acc_set_device_type (acc_device_nvidia);

      if (acc_get_device_type () != acc_device_nvidia)
	abort ();

      devnum = acc_get_num_devices (acc_device_host);
      if (devnum != 1)
	abort ();

      acc_shutdown (acc_device_nvidia);
    }

  if (acc_get_num_devices (acc_device_radeon))
    {
      acc_set_device_type (acc_device_radeon);

      if (acc_get_device_type () != acc_device_radeon)
	abort ();

      acc_shutdown (acc_device_radeon);

      acc_set_device_type (acc_device_radeon);

      if (acc_get_device_type () != acc_device_radeon)
	abort ();

      devnum = acc_get_num_devices (acc_device_host);
      if (devnum != 1)
	abort ();

      acc_shutdown (acc_device_radeon);
    }

  if (acc_get_device_type () == acc_device_default)
    abort ();

  return 0;
}
